File: Golang/NextPermutation/NextPermutation_test.go
package NextPermutation

import (
	"testing"
	"reflect"
)

func TestNextPermutation(t *testing.T) {

	array123 := []int{1, 2, 3}
	nextPermutation(array123)
	if reflect.DeepEqual(array123, []int{1, 3, 2}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	array321 := []int{3, 2, 1}
	nextPermutation(array321)
	if reflect.DeepEqual(array321, []int{1, 2, 3}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	array115 := []int{1, 1, 5}
	nextPermutation(array115)
	if reflect.DeepEqual(array115, []int{1, 5, 1}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/NextPermutationTest.py
from unittest import TestCase

from NextPermutation import NextPermutation


class TestNextPermutation(TestCase):
    def test_nextPermutation(self):
        np = NextPermutation()

        array123 = [1, 2, 3]
        np.nextPermutation(array123)
        self.assertEqual(array123, [1, 3, 2])

        array321 = [3, 2, 1]
        np.nextPermutation(array321)
        self.assertEqual(array321, [1, 2, 3])

        array115 = [1, 1, 5]
        np.nextPermutation(array115)
        self.assertEqual(array115, [1, 5, 1])
